### DISCUSSION TITLE
error in side reaction postprocessor

### DISCUSSION THREAD

**Discussion by zuckarm** [idaholab/moose#27715] [Q&A Modules: Phase field] [answered]
hi all,

i am trying to compute the reaction on side set 'bottom'. i am using the following code:
[Reaction]
type = SidesetReaction
direction = '0 -1 0'
stress_tensor = 'stress'
boundary = bottom
[]
where the 'stress' block is defined as:
[stress]
type = ComputeSmallDeformationStress
elasticity_model = elasticity
output_properties = 'stress'
outputs = exodus
[]

however, i am getting the following error:
The requested non-AD material property 'stress' of type 'RankTwoTensorTempl'
is already retrieved or declared as a AD property of type 'RankTwoTensorTempl'.

i used a material AD converter:
[sigma_yy]
type = MaterialADConverter
ad_props_in = stress
reg_props_out = reg_stress
boundary = bottom
[]

now i am getting a different error:
The requested AD material property 'stress' of type 'double'
is already retrieved or declared as a AD property of type 'RankTwoTensorTempl'.

can anyone suggest on how to resolve this?

thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

Is there an AD version of the material you could be using? Basically you are mixing AD and non AD here and that is not accepted 

You can use Debug/show_material_props to see which material is declaring and requesting each material property. 

Then make sure everything is consistently AD or not AD. 

For example you could use ADSidesetReaction

Guillaume

**Reply by zuckarm** [reply]
ADSidesetReaction worked, thanks!
